x-pack/plugins/alerting/server/task_runner/task_runner.ts

```diff
@@ -322,6 +322,11 @@ export class TaskRunner<
             includedHiddenTypes: ['alert', 'action'],
           });
 
+          const dataViews = await this.context.dataViews.dataViewsServiceFactory(
+            savedObjectsClient,
+            scopedClusterClient.asInternalUser
```

Comment:
Should we use `asInternalUser` or `asCurrentUser` here?

Reply:
And also should we use `scopedClusterClient` or `wrappedScopedClusterClient` here @ymao1 ?

Reply:
@dimaanj 

`wrappedScopedClusterClient` wraps the `search` function of `asInternalUser` and `asCurrentUser` to provide an abort signal and timing metrics. If you are using any other function, it shouldn't make a difference which one you use.

For `asInternalUser` vs `asCurrentUser`, it depends on your use case. Do you need it to respect the permissions of the API key that is associated with the rule? For example, if the user who created the rule doesn't have access to data views, should this data views service throw an error?

Reply:
Don't think we need it, since alerts shouldn't be available in read-only mode of Discover.